river/Touch.zig

```diff
@@ -102,6 +101,15 @@ fn handleTouchDown(
     self.wlr_cursor.absoluteToLayoutCoords(event.device, event.x, event.y, &lx, &ly);
 
     if (Cursor.surfaceAtPosition(lx, ly)) |result| {
+        switch (result.parent) {
+            .view => |view| {
+                self.seat.focusOutput(view.output);
+                self.seat.focus(view);
+                server.root.startTransaction();
+            },
+            .layer_surface => {},
+            .xwayland_unmanaged => assert(build_options.xwayland),
```

Comment:
This assert is not needed. I don't see how it will ever be reached unless XWayland support is enabled, in which case it will never fail.

Reply:
This is also just here because I copied it from somewhere else and wasn't sure what it's supposed to do so I left it.

Reply:
I think the assert is just used to assert that that case is never reached. I don't think it's needed in `checkFocusFollowsCursor` either, but it's perhaps just a "coding style" thing?

Reply:
```IRC
12:47   leon-p: ifreund: why is there an assert(build_options.xwayland) in checkFocusFollowsCursor()? If xwayland support is not enabled at compile time, that branch will never be reached anyway
14:05  ifreund: leon-p: exactly, if build_options.xwayland is false we want to mark the branch as unreachable
```